react/SearchResultFlexible.js

```diff
@@ -61,6 +61,50 @@ const SearchResultFlexible = ({
   thresholdForFacetSearch,
   lazyItemsRemaining,
 }) => {
+  const facetsFromPathname = pathname =>
+    pathname
+      .toLowerCase()
+      .split('/')
+      .filter(facet => facet)
+
+  useEffect(() => {
+    const pathname = window.location.pathname
+    const initialSearch = window.initialSearchFacets
+
+    if (!initialSearch && searchQuery.facets) {
+      window.initialSearchFacets = { ...searchQuery.facets, pathname }
+      return
+    }
+
+    if (!preventRouteChange) {
+      const initialFacets = facetsFromPathname(initialSearch.pathname)
+      const newFacets = facetsFromPathname(pathname)
+      const totalEqualFacets = newFacets.filter(facet =>
+        initialFacets.includes(facet)
+      ).length
+
+      if (
+        (newFacets.length < initialFacets.length &&
+          totalEqualFacets === newFacets.length) ||
+        (newFacets.length > initialFacets.length &&
+          totalEqualFacets === initialFacets.length)
+      ) {
+        return
+      }
+
+      window.initialSearchFacets = { ...searchQuery.facets, pathname }
+      return
+    }
+
+    if (
+      searchQuery.facets &&
+      initialSearch.pathname !== pathname &&
+      searchQuery.facets.queryArgs.query !== initialSearch.queryArgs.query
+    ) {
+      window.initialSearchFacets = { ...searchQuery.facets, pathname }
```

Comment:
Can we avoid storing this variable on `window`? We can store mutable variables in refs

Reply:
@klzns I tried to use context but it didn't work without the `preventRouteChange`, because the component is remounted and the context is lost. I think that with refs the behavior would be the same 😕

Reply:
Can we create a custom hook just to make it clearer of the intent of this hook?

Reply:
done @klzns, you can test [here](https://mobileclear--alssports.myvtex.com/clothing/jackets/Mens?map=c,c,specificationFilter_7721&__device=phone)